### PR TITLE
fix default values and type hints for GroupManager::search

### PR DESCRIPTION
**Pull request by icewind1991**
ints really are ints

Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: lib/private/Group/Manager.php [hunk 1]
```diff
@@ -252,7 +252,7 @@ public function search($search, $limit = null, $offset = null) {
 					$this->logger->debug('Group "' . $groupId . '" was returned by search but not found through direct access', ['app' => 'core']);
 				}
 			}
-			if (!is_null($limit) and $limit <= 0) {
+			if ($limit === 0) {
```
Comment (come-nc) [inline lib/private/Group/Manager.php:255]: Hum, the behavior when passing -1 explicitely changes then, no?

Comment (icewind1991) [reply lib/private/Group/Manager.php:255]: tbh, I don't know if the previous behavior with `-1` was intentional, and I can't find any code calling it with `-1` with a quick search

**Comment by icewind1991** [comment]
changed the `IGroupManager` to accept `null`s to prevent messing with code that relied on passing `null` as a default value, the `null`s are changed to the proper `-`/`0` before passing them to the backends so the backends still get the stronger typing

File: lib/private/Group/Database.php [hunk 2]
```diff
@@ -279,8 +279,12 @@ public function getGroups($search = '', $limit = null, $offset = null) {
 			)));
 		}
 
-		$query->setMaxResults($limit)
-			->setFirstResult($offset);
+		if (!$limit > 0) {
```
Comment (come-nc) [inline lib/private/Group/Database.php:282]:
```suggestion
		if ($limit <= 0) {
```

**Review by come-nc — APPROVED** [review]
See nitpick, fine otherwise.

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**